# Bacterial-Genomics-Sequencing-and-Bioinformatics-Training
# 🧬 Sequencing and Bioinformatics Training Course

Welcome to the official repository for the **Bacterial Genomics: Sequencing and Bioinformatics Training Course**. This one-week intensive training course is organized by the MicroHub Laboratory Kericho (MHK) to provide the participants with a comprehensive introduction to sequencing workflows and bioinformatics analysis, with a focus on bacterial genomics. The course is designed for laboratory scientists, researchers, and students who wish to understand the end-of-the-end process of Bacterial DNA sequencing and Bioinformatics analysis. Participants will engage in both hands-on wet-lab activities and practical bioinformatics sessions.

---

## 📅 Course Duration

**5 Days**  
**Time:** 8:00 AM – 5:00 PM  
**Breaks:**  
- Tea Break: 10:30 AM – 11:00 AM  
- Lunch: 1:00 PM – 2:00 PM

---

## 📘 Course Overview

This training combines hands-on wet-lab sessions and computational analyses to walk participants through the complete DNA sequencing workflow—from extraction to data interpretation. Participants will gain foundational skills in Unix, genome assembly, antimicrobial resistance gene detection, phylogenetics, and more.

---

## 🧪 Course Structure

| Day       | Topics                                                                 |
|-----------|------------------------------------------------------------------------|
| **Monday**    | Welcome, DNA Extraction (Lecture + Lab), DNA Quality Control         |
| **Tuesday**   | Library Preparation (Wet Lab)                                       |
| **Wednesday** | Finalize Library Prep, Start Sequencing, Intro to Unix & Conda      |
| **Thursday**  | Bacterial Genomics, QC, Genome Assembly, Species Typing             |
| **Friday**    | AMR & Virulence Screening, Phylogenetics, Online Tools, Feedback    |

---

## 🧠 Learning Outcomes

Participants will be able to:

- Extract and QC genomic DNA from bacterial samples  
- Prepare sequencing libraries and understand sequencing workflows  
- Use the Unix command line and manage environments with Conda  
- Retrieve, process, and analyze sequence datasets  
- Assemble genomes and identify species/strains  
- Detect antimicrobial resistance and virulence genes  
- Perform phylogenetic analysis using genomic data  
- Use online tools and databases (NCBI, CARD, PubMLST, etc.)

---

## 📂 Repository Contents


📁 course_material/

📄 [Course_manual.pdf](https://github.com/AMR-Bioinformatics/Bacterial-Genomics-Sequencing-and-Bioinformatics-Training/blob/main/Course_manual.pdf)

📁 Modules/

* [AMR and Virulence genes screening](https://github.com/AMR-Bioinformatics/Bacterial-Genomics-Sequencing-and-Bioinformatics-Training/tree/main/Modules/AMR%20and%20Virulence%20genes%20screening)
* [Introduction to Conda](https://github.com/AMR-Bioinformatics/Bacterial-Genomics-Sequencing-and-Bioinformatics-Training/tree/main/Modules/Introduction%20to%20Conda)
* [Introduction to Unix](https://github.com/AMR-Bioinformatics/Bacterial-Genomics-Sequencing-and-Bioinformatics-Training/tree/main/Modules/Introduction%20to%20Unix)
* [Sequence data QC and preprocessing](https://github.com/AMR-Bioinformatics/Bacterial-Genomics-Sequencing-and-Bioinformatics-Training/tree/main/Modules/Sequence%20data%20QC%20and%20preprocessing)
* [Species ID and bacterial typing](https://github.com/AMR-Bioinformatics/Bacterial-Genomics-Sequencing-and-Bioinformatics-Training/tree/main/Modules/Species%20ID%20and%20bacterial%20typing)

📁 Lectures/
* [DNA_Extraction.pdf]()
* [Library_Pre.pdf]()



## 🧑‍🏫 Trainers

[Dr Justin Nyasinga]() - Postdoctoral research fellow

[Collins Kigen](https://www.linkedin.com/in/collins-kigen-67b74910a/) - Head of Genomics and Bioinformatics

[Vanessa Natasha]() - Research Scientist, Genomics and Bionformatics

[James Wachira]() - Research Scientist, Genomics and Bionformatics
