# Phylogeny

<img width="2000" height="1700" alt="image" src="https://github.com/user-attachments/assets/5449e2d5-41fe-4f7d-a6e4-911de8400915" />

 Phylogeny is a pattern of historical evolutionary relationship among species and higher level taxa that is often presented as a tree diagram, or phylogenetic tree.
 The term phylogenetics is often applied to the study of such relationships.

 [Click here](https://github.com/AMR-Bioinformatics/Bioinformatics-Training/blob/main/Phylogenetics/Introduction.md) for more background on phylogeny

 - Phylogenetic methods are used to infer evolutionary relationships between organisms or genes.
 - These methods rely on analyzing various types of data, including molecular sequences, morphological characteristics, or behavioral traits, to reconstruct evolutionary history.
 - Key methods include distance-based, parsimony, maximum likelihood, and Bayesian approaches.

 1.  <ins>Distance-based methods</ins>:

These methods calculate pairwise distances between sequences and use clustering algorithms to build trees. Examples include Neighbor-Joining (NJ) and UPGMA

2. <ins>Parsimony methods</ins>:
   
These methods aim to find the tree with the fewest evolutionary changes (most parsimonious) to explain the observed data. 

3. <ins>Maximum Likelihood (ML) methods</ins>

These methods estimate the tree that is most likely to have produced the observed data, given a specific model of evolution.

4. <ins>Bayesian methods</ins>
These methods estimate the probability of a tree given the observed data and prior knowledge about evolutionary relationships.

Simple Phylogeny using parsnp
----
Parsnp is a command-line-tool for efficient microbial core genome alignment and SNP detection

#### Step 1 : Activate parsno and navigate to the working directory

```
Conda activate parsnp
mkdir phylogeny
cd phylogeny
```

#### Step 2: Download the training dataset

```
wget https://github.com/Natasha-Adongo/Bacterial-Genomics-Sequencing-and-Bioinformatics-Training/blob/main/Datasets/genomes_phylogeny.zip
```

#### Step 3: Download refrence genome

```
wget https://github.com/Natasha-Adongo/Bacterial-Genomics-Sequencing-and-Bioinformatics-Training/blob/main/Datasets/GCA_022494545.1_ASM2249454v1_genomic.fna

```

#### Step 4: Run parsnp

```
parsnp -r ./GCA_022494545.1_ASM2249454v1_genomic.fna -d ./genomes_phylogeny/*.fasta -o test_phylogeny_out

```
### Output files
<pre>
- parsnp.xmfa is the core-genome alignment.
- parsnp.ggr is the compressed representation of the alignment generated by the harvest-toolkit. This file can be used to visualize alignments with Gingr.
- parsnp.snps.mblocks is the core-SNP signature of each sequence in fasta format. This is the file which is used to generate parsnp.tree
- parsnp.tree is the resulting phylogeny.

</pre>


